osm2streets/src/intersection.rs

```diff
@@ -21,8 +18,7 @@ pub struct Intersection {
     pub point: Pt2D,
     /// This will be a placeholder until `Transformation::GenerateIntersectionGeometry` runs.
     pub polygon: Polygon,
-    pub complexity: IntersectionComplexity,
-    pub conflict_level: ConflictType,
+    pub t: IntersectionType,
```

Comment:
nitpick: Style wise it's constantly frustrating that `type` is a reserved word. We could use the special syntax `r#type` but I'd vote strongly against. Spelling out `intersection_type` is quite annoying. Tiny acronyms like `t` or `it` don't feel super readable. How do you feel about `kind`?

Reply:
I agree, and I really like `kind`.

Do we leave the name of the type itself alone, and work with the rust-ism that "type" is spelled "kind" in identifier contexts, or let rust have the term "type" all to itself, referring to the type system, and go with `IntersectionKind` (and `ConflictKind`)?

Reply:
I'm not too sure. :\ I think I've seen `Kind` used in structs in other projects. Up to you. For consistency, I'd slightly lean towards `IntersectionKind`

Reply:
I found a bunch of uses of  "kind" used in exactly this way in rust docs, so I'm convinced this is a better term than "type" across the board.